### PR TITLE
CLOUDP-60028: Implement index creation in atlas client

### PR DESCRIPTION
**Pull request by gssbzn**

### CODE REVIEW

**Comment by gssbzn** [comment]
@andreaangiolillo fixed some copy/pasta

**Comment by gssbzn** [comment]
@themantissa ready for you and your team

File: mongodbatlas/indexes.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+// IndexesServiceOp handles communication with the Cluster related methods
+// of the MongoDB Atlas API
+type IndexesServiceOp struct {
+	Client RequestDoer
+}
+
+var _ IndexesService = &IndexesServiceOp{}
+
+// IndexConfiguration represents current value of the metric that triggered the alert. Only present for alerts of type HOST_METRIC.
```
Comment (themantissa) [inline mongodbatlas/indexes.go:26]: Just a nit but this comment is confusing - this has nothing to do with metrics, yes?

**Comment by themantissa** [comment]
@marinsalinas @PacoDw @coderGo93 - can one of you give a sanity check review?   Sadly it's unlikely we'll use this w/ Terraform as the post with no response won't work well with the Terraform way.

File: mongodbatlas/indexes.go (hunk 1 above)
Comment (gssbzn) [reply mongodbatlas/indexes.go:26]: oops, probably copy/pasta, I'll fix this

**Pull request merged**